.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y cmake build-essential \
+          libboost-all-dev libeigen3-dev libfftw3-dev
+
+    - name: Build and install segyio
+      run: |
+        git clone --branch 1.9.12 https://github.com/equinor/segyio.git
+        cd segyio
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DBUILD_PYTHON=OFF -DEXPERIMENTAL=ON
+        make
+        sudo make install
+
+    - name: Build and test
+      run: |
+        mkdir build
+        cd build
+        cmake ..
+        make
+        ctest --output-on-failure
```

Comment:
Just FYI :smiley_cat: 

I am not sure if you intended to run all the tests or just unit ones, but it seems there is more?

If you change here to verbose
`ctest -v`

you would see that for example impedance suite prints

> All tests passed (2 assertions in 1 test case)

This is the content of `impedance/test/impedance.cpp`.
However in cmake file there are some [integration tests](https://github.com/equinor/si4ti/blob/be48da350c33a990a9c1e7c3afe822144b4be888/impedance/CMakeLists.txt#L59) present too.

Same for the timeshift.

From travis code it seems like one needs to call `make diff` to run them?

Reply:
Good catch. I did not read the travis config carefully enough. I have added `make diff`. I have also added a new section that installs `segyio` using `pip` because `diff.py` requires it.